File: Python/cocina.py
import threading
from pedido import Pedido
from cocinero import Cocinero
import os

class Cocina:
    def __init__(self):
        self.pedidos = []
        self.pedidos_lock = threading.Lock()
        self.log_lock = threading.Lock()
        
        # Crear archivo log si no existe
        if not os.path.exists("log_pedidos.txt"):
            open("log_pedidos.txt", "w").close()
    
    def agregar_pedido(self, pedido):
        self.pedidos.append(pedido)
    
    def iniciar_cocineros(self, num_cocineros):
        cocineros = []
        for i in range(num_cocineros):
            cocinero = Cocinero(f"Cocinero {i+1}", self)
            cocineros.append(cocinero)
            cocinero.start()
        
        # Esperar a que todos los cocineros terminen
        for cocinero in cocineros:
            cocinero.join()

def main():
    # Crear la cocina
    cocina = Cocina()
    
    # Crear pedidos
    platos = [
        "Pasta Carbonara",
        "Pizza Margherita",
        "Ensalada César",
        "Sopa de Tomate",
        "Filete de Salmón",
        "Risotto de Champiñones"
    ]
    
    # Agregar pedidos a la cocina
    for i, plato in enumerate(platos, 1):
        pedido = Pedido(i, plato)
        cocina.agregar_pedido(pedido)
    
    # Iniciar 3 cocineros
    print("Iniciando la preparación de pedidos...")
    cocina.iniciar_cocineros(3)
    print("\nTodos los pedidos han sido procesados.")

if __name__ == "__main__":
    main()

File: Java/Cocina.java
import java.util.ArrayList;
import java.util.List;

public class Cocina {
    private final List<Pedido> pedidosPendientes;
    private final List<Cocinero> cocineros;
    private int pedidosProcesados = 0;
    private final int totalPedidos;

    public Cocina(int numCocineros, List<Pedido> pedidos) {
        this.pedidosPendientes = new ArrayList<>(pedidos);
        this.cocineros = new ArrayList<>();
        this.totalPedidos = pedidos.size();

        // Crear los cocineros
        for (int i = 1; i <= numCocineros; i++) {
            cocineros.add(new Cocinero("Cocinero-" + i, this));
        }
    }

    public synchronized Pedido tomarPedido() {
        if (pedidosPendientes.isEmpty()) {
            pedidosProcesados++;
            if (pedidosProcesados == cocineros.size()) {
                System.out.println("Todos los pedidos han sido procesados.");
            }
            return null;
        }
        return pedidosPendientes.remove(0);
    }

    public void comenzarPreparacion() {
        for (Cocinero cocinero : cocineros) {
            cocinero.start();
        }

        // Esperar a que todos los cocineros terminen
        for (Cocinero cocinero : cocineros) {
            try {
                cocinero.join();
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
                System.err.println("Error esperando a cocinero: " + e.getMessage());
            }
        }
    }

    public static void main(String[] args) {
        // Crear lista de pedidos
        List<Pedido> pedidos = new ArrayList<>();
        String[] platos = {"Pizza", "Hamburguesa", "Pasta", "Ensalada", "Sopa", "Paella"};
        
        for (int i = 0; i < platos.length; i++) {
            pedidos.add(new Pedido(i + 1, platos[i]));
        }

        // Crear cocina con 3 cocineros
        Cocina cocina = new Cocina(3, pedidos);
        
        // Comenzar la preparación de pedidos
        cocina.comenzarPreparacion();
    }
}